### PR TITLE
Bugfix 15/11/21 Large Molecule Manipulations

### PR DESCRIPTION
**Pull request by trisyoungs**
This PR addresses issues with the manipulation of the coordinates of large molecules, specifically being related to the calculation of the centre-of-geometry.

### CODE REVIEW

File: src/classes/box.cpp [hunk 1]
```diff
@@ -198,7 +198,11 @@ Vec3<double> Box::getReal(Vec3<double> r) const
 }
 
 /*
+<<<<<<< HEAD
  * Geometry Calculation
+=======
+ * Geometry
+>>>>>>> efb17328b... Add manipulator functions to Molecule, and calculate COG correctly for large (all) molecules.
```
Comment (rprospero) [inline src/classes/box.cpp:205]: This feels like a git merge that didn't got through properly

File: src/classes/box.h [hunk 2]
```diff
@@ -136,7 +136,11 @@ class Box
     virtual double minimumDistanceSquared(const Vec3<double> &r1, const Vec3<double> &r2) const = 0;
 
     /*
+<<<<<<< HEAD
      * Geometry Calculation
+=======
+     * Geometry
+>>>>>>> efb17328b... Add manipulator functions to Molecule, and calculate COG correctly for large (all) molecules.
```
Comment (rprospero) [inline src/classes/box.h:143]: Git merge again

File: src/classes/molecule.cpp [hunk 3]
```diff
@@ -65,6 +67,81 @@ void Molecule::updateAtoms(std::vector<Atom> &source)
...
+    }
+}
+
+// General manipulation function working on reassembled molecule
+void Molecule::traverseLocal(const Box *box, ManipulationFunction action)
+{
+    std::vector<bool> flags(atoms_.size(), false);
+    action(atoms_[0], atoms_[0]->r());
+    recurseLocal(flags, box, 0, action);
+}
```
Comment (rprospero) [inline src/classes/molecule.cpp:126]: Do we need the non-const version?  It seems to do exactly the same as the const one.

**Review by rprospero — APPROVED** [review]
A couple of minor points, but otherwise looks good.

File: src/classes/molecule.cpp (hunk 3 above)
Comment (trisyoungs) [reply src/classes/molecule.cpp:126]: I'm afraid so - sometimes the lambda is read-only on the `Atom`s and called from a local `const` function, whereas other times it's used for actual manipulations.

**Pull request merged**